semgrep-core/src/parsing/pfff/scala_to_generic.ml

```diff
@@ -410,11 +410,26 @@ and v_expr e : G.expr =
   | Lambda v1 ->
       let v1 = v_function_definition v1 in
       G.Lambda v1 |> G.e
-  | New (v1, v2) ->
-      let _v1 = v_tok v1 and v2 = v_template_definition v2 in
-      let cl = G.AnonClass v2 |> G.e in
-      (* TODO: transform in New if v_template_definition is simple *)
-      G.Call (cl, fb []) |> G.e
+  | New (v1, v2) -> (
+      let v1 = v_tok v1 and v2 = v_template_definition v2 in
+      match v2 with
```

Comment:
you could have done also the match on the AST_Scala.v2 instead of v2 after the call to v_template_definition.